src/kernel/scheduling/thread.c

```diff
@@ -34,6 +35,11 @@ thread_t* create_thread(char* thread_name, void (*_entry_point)(void *), void* a
     // The stack grow backward, so the pointer will be the end of the stack
     new_thread->stack = stack_pointer + THREAD_DEFAULT_STACK_SIZE;
     new_thread->execution_frame->rsp = (uint64_t) new_thread->stack;
+
+    if (parent_task != NULL) {
+        new_thread->parent_task = parent_task;
```

Comment:
I'd move this code into `add_thread_to_task`, since part of adding the thread is changing it's parent_task to be the new task. That way you dont have to remember to update the parent if you use this function elsewhere.